nova_plugin/keypair.py

```diff
@@ -163,6 +163,15 @@ def validate_path_owner(path):
                 private_key_path)
             ctx.logger.error('VALIDATION ERROR: ' + err)
             raise NonRecoverableError(err)
+        else:
+            err = 'private key directory {0} is not writable'
```

Comment:
This code can be simplified to something along the lines of 
```
# elif to if pk_exists
elif private_key_path and not os.access(os.path.dirname(private_key_path), os.W_OK):
    err = 'private key directory {0} is not writable'
    raise NonRecoverableError(err.format(private_key_path))
```
or am i missing something?